app/cmd/rpc/mesh/session.go

```diff
@@ -89,124 +87,494 @@ func (sn DispatchResponse) GetSupportedNodes() []string {
 	return nodes
 }
 
-type AppSessionCache struct {
-	PublicKey       string
-	Chain           string
-	Dispatch        *DispatchResponse
+// NodeSession - contains error/valid information for the node-session relation
+type NodeSession struct {
+	PubKey string
+	// todo: if we can figure out a way to check this, otherwise receive/notify relays until fullNode return evidence sealed.
+	// ^^ that could work with problem, just few "free relays"
 	RemainingRelays int64
+	RelayMeta       *pocketTypes.RelayMeta
+	Validated       bool
+	RetryTimes      int
 	IsValid         bool
 	Error           *SdkErrorResponse
+	Session         *Session
 }
 
-// getAppSession - call ServicerURL to get an application session using retrieve header
-func getAppSession(relay *pocketTypes.Relay, model interface{}) *SdkErrorResponse {
-	servicerNode := getServicerFromPubKey(relay.Proof.ServicerPubKey)
+func (ns *NodeSession) CountRelay() (*NodeSession, bool) {
```

Comment:
Why would you ever need to return the node session itself?

Reply:
concat functions? like sugar session just that.

Reply:
good one, cleaning it.